Modules/Loadable/Markups/MRML/vtkMRMLMarkupsDisplayNode.h

```diff
@@ -66,26 +70,44 @@ class  VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkMRMLMarkupsDisplayNode : public
                                    unsigned long /*event*/,
                                    void * /*callData*/ ) override;
 
-  /// Convenienve function for getting the displayable markups node
+  /// Convenience function for getting the displayable markups node
   vtkMRMLMarkupsNode* GetMarkupsNode();
 
-  /// Active item (item that the mouse is hovered over).
-  /// This propoerty is computed on-the-fly and saved to file.
-  vtkGetMacro(ActiveComponentType, int);
+  /// Active item (item that the mouse or other interaction context is hovered over).
+  /// This property is computed on-the-fly and saved to file.
+  /// \param context Name of the interaction context. By default it is empty string, meaning mouse
+  int GetActiveComponentType(std::string context="");
   enum ComponentType
     {
     ComponentNone = 0,
     ComponentControlPoint,
     ComponentCenterPoint,
     ComponentLine
     };
+  struct ComponentInfo
+    {
+    ComponentInfo()
+      {
+      this->Type = ComponentNone;
+      this->Index = -1;
+      }
+    int Type;
+    int Index;
+    };
+
+  /// Index of active item (item that the mouse or other interaction context is hovered over).
+  /// This property is computed on-the-fly and saved to file.
+  /// \param context Name of the interaction context. By default it is empty string, meaning mouse
+  int GetActiveComponentIndex(std::string context="");
 
-  /// Index of active item (item that the mouse is hovered over).
-  /// This propoerty is computed on-the-fly and saved to file.
-  vtkGetMacro(ActiveComponentIndex, int);
+  /// Set active component type and index for interaction context (empty by default, meaning mouse)
+  void SetActiveComponent(int componentType, int componentIndex, std::string context="");
 
-  /// Set active component type and index.
-  void SetActiveComponent(int componentType, int componentIndex);
+  /// Query if there is an active component for any interaction context
+  /// \return List of interaction context names that have active components
```

Comment:
docstring `\return List of interaction context names that have active components` should not be added here. It belongs to the next function

Reply:
Good catch